includes/class-wc-gateway-stripe.php

```diff
@@ -135,16 +135,25 @@ public function __construct() {
 		add_filter( 'woocommerce_settings_api_sanitized_fields_' . $this->id, [ $this, 'settings_api_sanitized_fields' ] );
 		add_filter( 'woocommerce_gateway_' . $this->id . '_settings_values', [ $this, 'update_onboarding_settings' ] );
 
+		add_action( 'woocommerce_stripe_updated', [ $this, 'maybe_migrate_the_card_setting' ] );
+
 		// Note: display error is in the parent class.
 		add_action( 'admin_notices', [ $this, 'display_errors' ], 9999 );
 	}
 
 	/**
 	 * Checks if gateway should be available to use.
 	 *
+	 * Note: This method checks if the "card" method is available because the main gateway is responsible for cards specifically.
+	 *
 	 * @since 4.0.2
 	 */
 	public function is_available() {
+
+		if ( 'no' === $this->get_option( 'cards_enabled' ) ) {
+			return false;
+		}
```

Comment:
This approach of changing the `is_available()` function of the main gateway object to return a value indicating if cards are available or not is basically the same approach we took in the UPE class. See here: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/deferred-intent/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L1453

There are some slight differences, however thats caused by the fact that the UPE approach has access to a card UPE object, but non-upe doesn't so we use a new option instead.